<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Predictive_Control_API</class>
 <widget class="QMainWindow" name="Predictive_Control_API">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>922</width>
    <height>736</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MPC Learning Toolbox</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">background-color: rgb(203, 203, 203);</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout" stretch="2,14">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="Title">
      <property name="styleSheet">
       <string notr="true">
QFrame{
border: 3px solid #000000;
	border-color: rgb(225, 225, 225);
border-bottom-color: rgb(220, 220, 220);
background-color: rgb(225, 225, 225);

}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="7,1,1">
       <item>
        <widget class="QLabel" name="Title_name">
         <property name="styleSheet">
          <string notr="true">QFrame{
	font: 18pt &quot;Times New Roman&quot;;
border: none;
}</string>
         </property>
         <property name="text">
          <string>Model Predictive Control Interactive Tool</string>
         </property>
         <property name="textFormat">
          <enum>Qt::AutoText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="Run_btn">
         <property name="maximumSize">
          <size>
           <width>65</width>
           <height>65</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">
QLabel{
background-color: rgb(255, 255, 255);

}</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset>
           <normaloff>img/Run.PNG</normaloff>img/Run.PNG</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>55</width>
           <height>55</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>86</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="Principal">
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);
</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="7,3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>9</number>
       </property>
       <item>
        <widget class="QFrame" name="Graphs">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QFrame" name="G1">
            <property name="styleSheet">
             <string notr="true">QFrame{
background-color: rgb(220, 220, 220);
}
QVBoxLayout{
border: none;
}</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QVBoxLayout" name="Planta"/>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="G2">
            <property name="styleSheet">
             <string notr="true">QFrame{
background-color: rgb(220, 220, 220);
}
QVBoxLayout{
border: none;
}</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <layout class="QVBoxLayout" name="Entradas"/>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="Options">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,5,7">
          <item>
           <widget class="QLabel" name="Options_name">
            <property name="styleSheet">
             <string notr="true">QFrame{
font: 700 12pt &quot;Times New Roman&quot;;
border: none;
border: 3px solid #000000;
border-color: rgb(225, 225, 225);
}</string>
            </property>
            <property name="text">
             <string>Options</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="Model_Frame">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <widget class="QLabel" name="Model_name">
               <property name="styleSheet">
                <string notr="true">QFrame{
font: 12pt &quot;Times New Roman&quot;;
text-decoration: underline;
}</string>
               </property>
               <property name="text">
                <string>Model Parameters</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Simulation_time_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Simtime_label">
                  <property name="text">
                   <string>Simulation Time (seconds)</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_simtime">
                  <property name="text">
                   <string>30</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Gain_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Gain_label">
                  <property name="text">
                   <string>Input Gain &quot;K&quot;</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Gain">
                  <property name="text">
                   <string>1</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Tau_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Tau_label">
                  <property name="text">
                   <string>Time Constant &quot;Tau&quot;</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Tau">
                  <property name="text">
                   <string>1</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Saturation_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Simtime_Saturation">
                  <property name="text">
                   <string>Input Up/Down Saturation</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Saturation">
                  <property name="text">
                   <string>-10,10</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="Controller_frame">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <widget class="QLabel" name="Controller_name">
               <property name="styleSheet">
                <string notr="true">QFrame{
font: 12pt &quot;Times New Roman&quot;;
text-decoration: underline;
}</string>
               </property>
               <property name="text">
                <string>Controller Parameters</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Reference_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_12" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Reference_label">
                  <property name="text">
                   <string>Reference</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Reference">
                  <property name="text">
                   <string>1,2,4,1</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Control_Period_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="2,2,1">
                <item>
                 <widget class="QLabel" name="Control_Period_label">
                  <property name="text">
                   <string>Controller Period</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSlider" name="horizontalSlider">
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                  <property name="pageStep">
                   <number>10</number>
                  </property>
                  <property name="value">
                   <number>10</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>0.1</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Horizon_Size_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Horizon_Size_label">
                  <property name="text">
                   <string>Moving Horizon Size</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Horizon_Size">
                  <property name="text">
                   <string>30</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Learning_Rate_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Learning_Rate_label">
                  <property name="text">
                   <string>Learning Rate</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Learning_Rate">
                  <property name="text">
                   <string>1e-2</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Step_decay_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Step_decay_label">
                  <property name="text">
                   <string>Step Decay (%/10 iter)</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Step_decay">
                  <property name="text">
                   <string>0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QFrame" name="Iterations_frame">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="3,1">
                <item>
                 <widget class="QLabel" name="Iterations_label">
                  <property name="text">
                   <string>Iterations</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="LineEdit_Iterations">
                  <property name="text">
                   <string>100</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
